src/Microsoft.DotNet.Tools.Run/RunCommand.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.Extensions.ProjectModel;
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.Tools.Run
+{
+    public class RunCommand
+    {
+        public string Framework = null;
+        public string Configuration = null;
+        public bool PreserveTemporary = false;
+        public string Project = null;
+        public IReadOnlyList<string> Args = null;
+
+        bool _isInteractive = false;
+        ProjectContext _context;
+        List<string> _args;
+
+        public int Start()
+        {
+            CalculateDefaultsForNonAssigned();
+
+            if (_isInteractive)
+            {
+                return RunInteractive(Project);
+            }
+            else
+            {
+                return RunExecutable();
+            }
+        }
+
+        private void CalculateDefaultsForNonAssigned()
+        {
+            if (!string.IsNullOrEmpty(Project))
+            {
+                if (File.Exists(Project) && (Path.GetExtension(Project) == ".csx"))
+                {
+                    _isInteractive = true;
```

Comment:
Would this check be more appropriate in the parent function? Doesn't seem to have much to do with calculating defaults.